Source/IGListAdapter.m

```diff
@@ -759,6 +760,23 @@ - (__kindof UICollectionReusableView *)dequeueReusableSupplementaryViewOfKind:(N
     return [collectionView dequeueReusableSupplementaryViewOfKind:elementKind withReuseIdentifier:identifier forIndexPath:indexPath];
 }
 
+- (__kindof UICollectionReusableView *)dequeueReusableSupplementaryViewOfKind:(NSString *)elementKind
+                                                         forSectionController:(IGListSectionController <IGListSectionType> *)sectionController
+                                                                      nibName:(NSString *)nibName
+                                                                       bundle:(NSBundle *)bundle
+                                                                      atIndex:(NSInteger)index {
+    IGAssertMainThread();
```

Comment:
Let's also add:
```objc
IGParameterAssert([nibName length] > 0);
IGParameterAssert([elementKind length] > 0);
```
I realize those are also missing from other methods. Adding some starter tasks for that.